### PR TITLE
8292267: Clean up synchronizer.hpp

### PR DESCRIPTION
**Pull request by iklam**
Fix two problems with synchronizer.hpp:

- It's unnecessarily included in two popular headers: monitorChunk.hpp and frame_\<cpu\>.hpp. The latter is most problematic because it includes synchronizer.hpp in the middle of a class declaration!
- synchronizer.hpp includes linkedlist.hpp. This can be avoid by moving the definition of Synchronizer::PtrList into synchronizer.cpp.

These headers are included much less after this PR (out of ~1000 hotspot .o files)

```
linkedlist.hpp    857 -> 101
synchronizer.hpp  848 -> 122
resourceHash.hpp  850 -> 270
```

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8292267](https://bugs.openjdk.org/browse/JDK-8292267): Clean up synchronizer.hpp

### Reviewers
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**) ⚠️ Review applies to [9b372650](https://git.openjdk.org/jdk/pull/9849/files/9b372650c147c9c86cb6ac4058f7eda3ccbbd26e)
 * [Daniel D. Daugherty](https://openjdk.org/census#dcubed) (@dcubed-ojdk - **Reviewer**) ⚠️ Review applies to [9b372650](https://git.openjdk.org/jdk/pull/9849/files/9b372650c147c9c86cb6ac4058f7eda3ccbbd26e)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to [9b372650](https://git.openjdk.org/jdk/pull/9849/files/9b372650c147c9c86cb6ac4058f7eda3ccbbd26e)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9849/head:pull/9849` \
`$ git checkout pull/9849`

Update a local copy of the PR: \
`$ git checkout pull/9849` \
`$ git pull https://git.openjdk.org/jdk pull/9849/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9849`

View PR using the GUI difftool: \
`$ git pr show -t 9849`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9849.diff">https://git.openjdk.org/jdk/pull/9849.diff</a>

</details>

### CODE REVIEW

**Review by coleenp — APPROVED** [review]
Very nice.

File: src/hotspot/share/gc/shenandoah/shenandoahPadding.hpp [hunk 1]
```diff
@@ -25,6 +25,8 @@
 #ifndef SHARE_GC_SHENANDOAH_SHENANDOAHPADDING_HPP
 #define SHARE_GC_SHENANDOAH_SHENANDOAHPADDING_HPP
 
+#include "memory/padded.hpp"
```
Comment (dcubed-ojdk) [inline src/hotspot/share/gc/shenandoah/shenandoahPadding.hpp:28]: This one surprised me, but I'm guessing it was included indirectly
from synchronizer.hpp via some frame header...

File: src/hotspot/share/prims/jvmtiExport.hpp [hunk 2]
```diff
@@ -47,7 +47,7 @@ class JvmtiEventControllerPrivate;
 class JvmtiManageCapabilities;
 class JvmtiEnv;
 class JvmtiThreadState;
-
+class ThreadsList;
```
Comment (dcubed-ojdk) [inline src/hotspot/share/prims/jvmtiExport.hpp:50]: Not in sorted order relative to OopStorage. Does it need to be?

**Review by dcubed-ojdk — APPROVED** [review]
Looks great. I only posted a few nit comments.

**Review by dholmes-ora — APPROVED** [review]
Looks fine.

Thanks.

**Comment by iklam** [comment]
Thanks @coleenp @dholmes-ora @dcubed-ojdk for the review.
Tiers 1,2 and build-tier5 passed on top of #9927
/integrate

**Pull request closed without merging**